File: for her.py

import pygame
import random
import math

pygame.init()
WIDTH, HEIGHT = 700, 400
screen = pygame.display.set_mode((WIDTH, HEIGHT))
pygame.display.set_caption("Te Amo Michi")
clock = pygame.time.Clock()

heart_center = (WIDTH // 2, HEIGHT // 2 - 50)
particles = []

def heart_shape(t, scale=100):
    x = 16 * math.sin(t) ** 3
    y = 13 * math.cos(t) - 5 * math.cos(2 * t) - 2 * math.cos(3 * t) - math.cos(4 * t)
    return (x * scale, -y * scale)

def generate_heart_particles(n=3000):
    result = []
    for _ in range(n):
        while True:
            t = random.uniform(0, 2 * math.pi)
            radius = random.uniform(0.2, 1.0)  # fill the heart
            x, y = heart_shape(t, scale=radius * 7)
            x += heart_center[0]
            y += heart_center[1]
            if 0 < x < WIDTH and 0 < y < HEIGHT:
                result.append({
                    "pos": [x, y],
                    "vel": [0, 0],
                    "base": [x, y]
                })
                break
    return result

def draw_text(surface, text, pos, size, color):
    font = pygame.font.SysFont("Arial", size, bold=True)
    text_surface = font.render(text, True, color)
    rect = text_surface.get_rect(center=pos)
    surface.blit(text_surface, rect)

particles = generate_heart_particles()

running = True
while running:
    screen.fill((0, 0, 0))
    mx, my = pygame.mouse.get_pos()

    for p in particles:
        px, py = p["pos"]
        bx, by = p["base"]
        dx = mx - px
        dy = my - py
        dist_mouse = math.hypot(dx, dy)
        if dist_mouse < 25:
            angle = math.atan2(dy, dx)
            force = (25 - dist_mouse) * 0.05
            p["vel"][0] -= math.cos(angle) * force
            p["vel"][1] -= math.sin(angle) * force

        # pull to base
        p["vel"][0] += (bx - px) * 0.01
        p["vel"][1] += (by - py) * 0.01

        # friction
        p["vel"][0] *= 0.9
        p["vel"][1] *= 0.9

        p["pos"][0] += p["vel"][0]
        p["pos"][1] += p["vel"][1]

        # gradient based on distance from center
        d = math.hypot(p["pos"][0] - heart_center[0], p["pos"][1] - heart_center[1])
        t = min(d / 140, 1)
        r = int(255)
        g = int(128 * (1 - t))
        b = int(160 * (1 - t))
        pygame.draw.circle(screen, (r, g, b), (int(p["pos"][0]), int(p["pos"][1])), 2)

    draw_text(screen, "TE AMO MICHI", (WIDTH // 2, HEIGHT - 50), 48, (255, 0, 0))
    pygame.display.flip()

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    clock.tick(60)

pygame.quit()

File: FOR HER.js
<!DOCTYPE html>
<html lang="es">
<head>
  <meta charset="UTF-8" />
  <title>Te Amo Michi</title>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/p5.js/1.6.0/p5.min.js"></script>
  <style>
    body {
      margin: 0;
      overflow: hidden;
      background-color: black;
    }
  </style>
</head>
<body>
<script>
let particles = [];
let basePositions = [];
let heartCenter;

function setup() {
  createCanvas(windowWidth, windowHeight);
  heartCenter = createVector(width / 2, height / 2 - 50);
  let total = 3000; // más partículas para rellenar

  // Generar partículas dentro del corazón
  for (let i = 0; i < total; i++) {
    let pos = getRandomHeartPoint();
    basePositions.push(pos.copy());
    particles.push({
      pos: pos.copy(),
      vel: createVector(0, 0)
    });
  }
}

function draw() {
  background(0);

  for (let i = 0; i < particles.length; i++) {
    let p = particles[i];
    let base = basePositions[i];

    let d = dist(mouseX, mouseY, p.pos.x, p.pos.y);
    if (d < 100) {
      let angle = atan2(p.pos.y - mouseY, p.pos.x - mouseX);
      let force = (100 - d) * 0.05;
      let push = p5.Vector.fromAngle(angle).mult(force);
      p.vel.add(push);
    }

    let pull = p5.Vector.sub(base, p.pos).mult(0.05);
    p.vel.add(pull);
    p.vel.mult(0.9);
    p.pos.add(p.vel);

    // Color degradado radial: más rosado lejos del centro, rojo en el centro
    let distToCenter = dist(p.pos.x, p.pos.y, heartCenter.x, heartCenter.y);
    let maxDist = 140; // radio del corazón
    let t = constrain(distToCenter / maxDist, 0, 1);
    let r = lerp(255, 255, t);  // de rosado a rojo
    let g = lerp(128, 0, t);
    let b = lerp(160, 0, t);
    fill(r, g, b);
    noStroke();
    circle(p.pos.x, p.pos.y, 2);
  }

  drawText();
}

function drawText() {
  textAlign(CENTER, CENTER);
  textSize(48);
  for (let dx = -2; dx <= 2; dx++) {
    for (let dy = -2; dy <= 2; dy++) {
      if (dx !== 0 || dy !== 0) {
        fill(255, 0, 0, 50);
        text("TE AMO MICHI", width / 2 + dx, height - 80 + dy);
      }
    }
  }
  fill(255, 0, 0);
  text("TE AMO MICHI", width / 2, height - 80);
}

// Generador de puntos dentro de un corazón (usando fórmula implícita)
function getRandomHeartPoint() {
  while (true) {
    let x = random(-1, 1);
    let y = random(-1.5, 1.1);
    let fx = 16 * pow(x, 3);
    let fy = 13 * y - 5 * cos(2 * x * PI) - 2 * cos(3 * x * PI) - cos(4 * x * PI);
    let d = pow(x * 16 * sin(x * PI), 3) + pow(fy, 2);

    // Forma implícita del corazón (simplificada)
    let r = sqrt(pow(x, 2) + pow(y, 2));
    let heartShape = pow(x * x + y * y - 1, 3) - x * x * y * y * y;
    if (heartShape <= 0) {
      let scale = 100; // tamaño más pequeño
      return createVector(x * scale + width / 2, -y * scale + height / 2 - 50);
    }
  }
}
</script>
</body>
</html>
